torch_xla/csrc/runtime/pjrt_registry.cc

```diff
@@ -92,29 +115,27 @@ InitializePjRt(const std::string& device_type) {
     int global_process_rank = sys_util::GetEnvInt("RANK", local_process_rank);
     int local_world_size = sys_util::GetEnvInt("LOCAL_WORLD_SIZE", 1);
     int global_world_size = sys_util::GetEnvInt("WORLD_SIZE", local_world_size);
-    std::string master_addr =
-        runtime::sys_util::GetEnvString("MASTER_ADDR", "localhost");
-    std::string port = runtime::sys_util::GetEnvString(
-        "XLA_COORDINATOR_PORT", XlaCoordinator::kDefaultCoordinatorPort);
-
-    bool spmd = sys_util::GetEnvBool("XLA_USE_SPMD", false);
-    std::optional<std::set<int>> allowed_devices;
-    if (!spmd) {
-      allowed_devices = std::set{local_process_rank};
-    }
 
     std::shared_ptr<xla::KeyValueStoreInterface> kv_store;
-    if (global_world_size > 1) {
-      // Use the distributed key-value store from DistributedRuntimeClient.
-      coordinator = std::make_unique<XlaCoordinator>(
-          global_process_rank, global_world_size, master_addr, port);
-      std::shared_ptr<xla::DistributedRuntimeClient> distributed_client =
-          coordinator->GetClient();
-      kv_store = xla::GetDistributedKeyValueStore(distributed_client,
-                                                  /*key_prefix=*/"gpu:");
+    std::optional<std::set<int>> allowed_devices;
+    bool spmd = sys_util::GetEnvBool("XLA_USE_SPMD", false);
+    if (spmd) {
+      if (global_world_size > 1) {
```

Comment:
After playing around following the last pin update in #6289, I think the XLA coordinator might actually be required in every case.

Reply:
The script hangs even if I modify the code to create a XLA coordinator in every case:
```
import torch
import torch_xla
import torch_xla.runtime as xr
import torch_xla.core.xla_model as xm

print(xm.xla_real_devices(xm.get_xla_supported_devices()))
print(xm.xla_device())

import torch

print(torch.randn((3, 3), device=xm.xla_device()))
```

Reply:
Yeah, this seems to happen when there are multiple GPUs in the current process and try to execute on one.

Reply:
fixed by a pin update.